src/functionalities/InputNode.js

```diff
@@ -6,11 +8,38 @@ class InputNode extends Functionality {
     return 'InputNode'
   }
 
-  constructor(functionalityData) {
+  static get VALID_SOURCES() {
+    return {
+      OSPIN_WEBAPP: { name: 'ospin-webapp' },
+      UNSPECIFIED: { name: 'unspecified' },
+    }
+  }
+
+  static get VALID_SOURCE_NAMES() {
+    return Object.values(InputNode.VALID_SOURCES).map(({ name }) => name)
+  }
+
+  static get DEFAULT_SOURCE() {
+    return InputNode.VALID_SOURCES.UNSPECIFIED
+  }
+
+  static get SCHEMA() {
+    return Joi.object({
+      source: Joi.object({
+        name: Joi.string().allow(...InputNode.VALID_SOURCE_NAMES).required(),
+      }),
+    }).concat(super.SCHEMA)
```

Comment:
your tip yesterday to invert this call, so the `super.SCHEMA` comes first, was very good and probably what we want to do across the board going forward. Then the base classes won't have the opportunity to overwrite the children SCHEMA props.